src/toil/utils/toilDebugFile.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (C) 2017- Regents of the University of California
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Debug tool for copying files contained in a toil jobStore.
+"""
+
+from __future__ import absolute_import
+import logging
+import fnmatch
+import os.path
+
+from toil.lib.bioio import getBasicOptionParser
+from toil.lib.bioio import parseBasicOptions
+from toil.common import Toil, jobStoreLocatorHelp, Config
+from toil.version import version
+
+logger = logging.getLogger( __name__ )
+
+def recursiveGlob(directoryname, glob_pattern):
+    '''
+    Walks through a directory and its subdirectories looking for files matching
+    the glob_pattern and returns a list=[].
+
+    :param directoryname: Any accessible folder name on the filesystem.
+    :param glob_pattern: A string like "*.txt", which would find all text files.
+    :return: A list=[] of absolute filepaths matching the glob pattern.
+    '''
+    directoryname = os.path.abspath(directoryname)
+    matches = []
+    for root, dirnames, filenames in os.walk(directoryname):
+        for filename in fnmatch.filter(filenames, glob_pattern):
+            absolute_filepath = os.path.join(root, filename)
+            matches.append(absolute_filepath)
+    return matches
+
+def fetchJobStoreFiles(jobStore, options):
+    """
+    Takes a list of file names as glob patterns, searches for these within a
+    given directory, and attempts to take all of the files found and copy them
+    into options.localFilePath.
+
+    :param jobStore: A fileJobStore object.
+    :param options.fetch: List of file glob patterns to search
```

Comment:
This docstrings are really nice and helpful! This is a minor complaint: PyCharm for me complains though because it expects four parameters when you are actually just describing all of the properties of options.